jupyterlab_gitsync/src/service/git.ts

```diff
@@ -0,0 +1,80 @@
+import { requestAPI } from './request_api';
+import { ISignal, Signal } from '@lumino/signaling';
+
+/**
+ *
+ * Class responsible for requesting git handlers on the server side
+ * to synchronize local repository with remote repository
+ *
+ */
+
+export class GitManager {
+  // Member Fields
+  private _path: string;
+  private _executablePath: string;
+  private _options: string[] = [];
+  private _fileConflicts: string[] = undefined;
+  private _syncCompleted: Signal<this, void> = new Signal<this, void>(this);
+  private _mergeConflict: Signal<this, void> = new Signal<this, void>(this);
+
+  constructor(path: string, options?: {remote: string, worktree:string}) {
+    this._setup(path);
+    if (options){
+      this._options[0] = options.remote;
+      this._options[1] = options.worktree;
+    }
+  }
+
+  get path(): string {
+    return this._path;
+  }
+
+  get fileConflicts(): string[] {
+    return this._fileConflicts;
+  }
+
+  get syncCompleted(): ISignal<this, void> {
+    return this._syncCompleted;
+  }
+
+  get mergeConflict(): ISignal<this, void> {
+    return this._mergeConflict;
+  }
+
+  async sync(args: string = null, emit = true) {
```

Comment:
It doesn't look like either of these method arguments are used.

Reply:
This should be a TODO item. I'll add a comment to make that more clear.